utils/order-utils/test/orders.js

```diff
@@ -168,6 +172,122 @@ describe('Orders', async () => {
     assert.equal(errors[0], 'signer allowance is too low')
   })
 
+  it('Check NFT order without balance or allowance', async () => {
+    const order = await orders.getOrder({
+      expiry: '1604787494',
+      nonce: '101',
+      signer: {
+        wallet: KNOWN_GANACHE_WALLET,
+        token: ASTAddress,
+        param: '0',
+        kind: ERC20_INTERFACE_ID,
+      },
+      sender: {
+        wallet: senderWallet,
+        token: cryptoKittiesAddress,
+        param: '460',
+        kind: ERC721_INTERFACE_ID,
+      },
+    })
+
+    order.signature = await signatures.getWeb3Signature(
+      order,
+      KNOWN_GANACHE_WALLET,
+      rinkebySwap,
+      GANACHE_PROVIDER
+    )
+
+    const errors = await orders.checkOrder(order, 'rinkeby')
+    assert.equal(errors.length, 2)
+    assert.equal(errors[0], "sender doesn't own NFT")
+    assert.equal(errors[1], 'sender no NFT approval')
+  })
+
+  it('Check invalid token kind', async () => {
+    const order = await orders.getOrder({
+      expiry: '1604787494',
+      nonce: '101',
+      signer: {
+        wallet: KNOWN_GANACHE_WALLET,
+        token: ASTAddress,
+        param: '0',
+        kind: ERC20_INTERFACE_ID,
+      },
+      sender: {
+        wallet: senderWallet,
+        token: cryptoKittiesAddress,
+        param: '460',
+        kind: 'INVALID_KIND',
+      },
+    })
+
+    order.signature = await signatures.getWeb3Signature(
+      order,
+      KNOWN_GANACHE_WALLET,
+      rinkebySwap,
+      GANACHE_PROVIDER
+    )
+
+    const errors = await orders.checkOrder(order, 'rinkeby')
+    assert.equal(errors.length, 1)
+    assert.equal(errors[0], 'sender token kind invalid')
+  })
+
+  it('Check NFT order without allowance', async () => {
+    const order = await orders.getOrder({
+      expiry: '1604787494',
+      nonce: '101',
+      signer: {
+        wallet: KNOWN_GANACHE_WALLET,
+        token: ASTAddress,
+        param: '0',
+        kind: ERC20_INTERFACE_ID,
+      },
+      sender: {
+        wallet: kittyWallet,
+        token: cryptoKittiesAddress,
+        param: '460',
+        kind: ERC721_INTERFACE_ID,
+      },
+    })
+
+    order.signature = await signatures.getWeb3Signature(
+      order,
+      KNOWN_GANACHE_WALLET,
+      rinkebySwap,
+      GANACHE_PROVIDER
+    )
+
+    const errors = await orders.checkOrder(order, 'rinkeby')
+    assert.equal(errors.length, 1)
+    assert.equal(errors[0], 'sender no NFT approval')
+  })
+
+  it('Check NFT order to a contract', async () => {
+    const order = await orders.getOrder({
+      expiry: '1604787494',
```

Comment:
can we capture all instances of `1604787494` and put it at the top as a const so that we know it's reference to the ganache time `2017-05-10T00:00:00+00:00`?

Reply:
Its actually `Saturday, 7 November 2020 22:18:14`
But yes will update accordingly

Reply:
This is one of my laptop bookmarks lol
https://www.epochconverter.com/